### PR TITLE
add a troubleshooting document

### LINKED ISSUE #36: write a troubleshooting document
Should cover 3 most common problems:
1) collector does not respond
2) collector fails to communicate with some/all Docker controllers
3) other runtime errors

### PR DESCRIPTION
**Pull request by EranGabber**
Closes #36.

### CODE REVIEW

File: TROUBLESHOOTING.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    5. If the container `kubernetes/cluster-insight` is listed as running,
+       then it is probably stuck.
+       Read its log by the command:
+       `sudo docker log CONTAINER_ID`. Pay attention to the messages at the
+       tail of the log.
+       The CONTAINER_ID is a 12-digit hexadecimal number. It is
+       listed in the first column of the output of `sudo docker ps`.
+    6. Stop the running container with:
+       `sudo docker stop CONTAINER_ID` and then delete it with:
+       `sudo docker rm CONTAINER_ID`.
```
Comment (supriyagarg) [inline TROUBLESHOOTING.md]: `sudo docker rm -f CONTAINER_ID` is a shortcut for the two commands.

File: TROUBLESHOOTING.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+## The server fails with a different run-time failure
+This may be caused by a genuine bug.
+
+1.  Read the log of the running Cluster-Insight server and look for additional
+    information about the failure. Follow these instructions:
+    1. Login to the master node running the Cluster-Insight server.
+    2. Find the container ID of the Cluster-Insight server with the command:
+       `sudo docker ps`
+    3. Read the logs of the Cluster-Insight server with the command:
+       `sudo docker log CONTAINER_ID`
```
Comment (supriyagarg) [inline TROUBLESHOOTING.md]: Same as before - `sudo docker logs CONTAINER_ID`

File: TROUBLESHOOTING.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    or by the following command:
+    `gcloud compute --project=PROJECT_ID instances list`.
+    If the IP address changed, use the new one in your browser.
+
+3.  Is the firewall rule correct?
+    If you have used this Cluster-Insight server in the past, it is unlikely
+    that the firewall rule got corrupt.
+
+    List the firewall rules of the cluster on GCP with:
+    `gcloud compute --project=PROHECT_ID firewall-rules list`
```
Comment (supriyagarg) [inline TROUBLESHOOTING.md]: PROHECT_ID -> PROJECT_ID

File: TROUBLESHOOTING.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    4. If the container `kubernetes/cluster-insight` is not listed in the
+       output of `sudo docker ps`, you will have to start it.
+       Please follow the instructions in the [README](README.md) file.
+       You may need to
+       delete a previous dead container with the same name if the `docker run`
+       command fails.
+    5. If the container `kubernetes/cluster-insight` is listed as running,
+       then it is probably stuck.
+       Read its log by the command:
+       `sudo docker log CONTAINER_ID`. Pay attention to the messages at the
```
Comment (supriyagarg) [inline TROUBLESHOOTING.md]: Should be `sudo docker logs CONTAINER_ID` (`logs -f` is better in order to see the results of the log on hitting the API)

File: TROUBLESHOOTING.md (hunk 1 above)
Comment (EranGabber) [inline TROUBLESHOOTING.md]: Done.

File: TROUBLESHOOTING.md (hunk 2 above)
Comment (EranGabber) [inline TROUBLESHOOTING.md]: Done.

File: TROUBLESHOOTING.md (hunk 3 above)
Comment (EranGabber) [inline TROUBLESHOOTING.md]: Done.

File: TROUBLESHOOTING.md (hunk 4 above)
Comment (EranGabber) [inline TROUBLESHOOTING.md]: Done.

**Comment by EranGabber** [comment]
I addressed your comments and added more information about debugging Docker daemon problems. PTAL.

**Pull request merged**